src/external/VTK.jl

```diff
@@ -0,0 +1,79 @@
+# Uses WriteVTK.jl to convert various structures and matrices to VTk file format
+using WriteVTK
+
+"""
+    EnergyToVTK(VTKFileName::String, energy::Energies{::Number})
+
+Stores the Energy structure in a VTK image file and returns the name of the file.
+"""
+function EnergyToVTK(VTKFileName::String, energy::Energies{<:Number})
```

Comment:
I noticed some code as in [this function](https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/external/etsf_nanoquanta.jl#L19) which were in Camel Case.
Apart from changing the naming convention to Pascal type, you wanted some changes in the function parameter types. I am not sure about how much generalization would be good here. You mentioned DFTK structures are meant to changed in the future so annotating types like `AbstractMatrix` for matrix (like densities) will be okay? or, should I not annotate types for non-structures?

Reply:
Or maybe if we are putting all these in a single VTK(as you mentioned) using single function we won't need such type specifications anymore.

Reply:
Yes the exception to the snail-case rule in DFTK (much as in base Julia) are constructor-like functions, such as the one you are pointing out.

As a general rule and guideline, I would not put types at first unless you need it to dispatch. In a later pass (when the structure of the code is more clear and thus so are the public interfaces), then I tend to add the types. So in line with that I would follow your second comment and not worry that much about the types for the redesign.

Reply:
I am almost done with new set of changes, I just have a question. If I am not wrong, the  wave function will be stored in real space rather than the Fourier space. The waves in real space have imaginary components, (You mentioned that imaginary parts are more difficult to interpret visually) I am finding it difficult to decide whether to store the modulus of the number or the real part(trimming out the imaginary part).

Reply:
Sorry that I ignored your question for so long. Yes, the way you are doing it now the wave function is stored in real space. More precisely only the real part of the real-space version of the wave function is stored (due to your `real.( ...  )`), which is an issue as the imaginary part also contains useful information (the wave function is complex-valued).

Be a little careful with the terminology here: Unfortunately "real" has two different meanings in this context. There is the real part of the complex-valued wave function and then there is the real-space and the fourier-space (or reciprocal-space) representation of said wave function. Both of which are again complex-valued. So in fact both in real space and in reciprocal space you have a real and an imaginary part and both are needed.